File: src.JAVA/src/swa/ariadne/ctrl/SolverState.java
package swa.ariadne.ctrl;

/**
 * The states an {@link AriadneController} may be in.
 * 
 * @author Stephan.Wacker@web.de
 */
public enum SolverState
{
    /** A maze has been created.  No timer is active. */
    Ready,
    
    /** A solver has been created.  The step timer is active. */
    Running,
    
    /** A solver has been created.  The step timer is inactive. */
    Paused,
    
    /** The current maze is solved. */
    Finished,

    /** A maze has been created.  The repeat timer is active. */
    ReadyAndScheduled,

    /** The current maze is solved.  The repeat timer is active. */
    FinishedAndScheduled,
}

File: src.CS/SWA.Ariadne.Logic/SolverState.cs
namespace SWA.Ariadne.Logic
{
    /// <summary>
    /// The states an AriadneController may be in.
    /// </summary>
    public enum SolverState
    {
        /// <summary>
        /// A maze has been created.  No timer is active.
        /// </summary>
        Ready = 1,

        /// <summary>
        /// A solver has been created.  The step timer is active.
        /// </summary>
        Running,

        /// <summary>
        /// A solver has been created.  The step timer is inactive.
        /// </summary>
        Paused,

        /// <summary>
        /// The current maze is solved.
        /// </summary>
        Finished,

        /// <summary>
        /// Variant of the basic states, when the repeat timer is active.
        /// </summary>
        Scheduled = 16,

        /// <summary>
        /// A maze has been created.  The repeat timer is active.
        /// </summary>
        ReadyAndScheduled = Ready | Scheduled,

        /// <summary>
        /// The current maze is solved.  The repeat timer is active.
        /// </summary>
        FinishedAndScheduled = Finished | Scheduled,
    }
}

